File: fixtures/flutter-basic/lib/screens/detail_screen.dart
import 'package:flutter/material.dart';

import '../widgets/product_card.dart';

/// DetailScreen — 커스텀 컴포넌트 합성 화면.
/// 아키타입 2: ProductCard(PriceTag 내포)를 3회 사용 — 2단 깊이 인라이닝 검증용.
class DetailScreen extends StatelessWidget {
  const DetailScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFFF3EFF4),
      appBar: AppBar(
        backgroundColor: const Color(0xFF6750A4),
        foregroundColor: Colors.white,
        title: const Text(
          'Product Catalog',
          style: TextStyle(fontWeight: FontWeight.bold, fontSize: 20),
        ),
        leading: const BackButton(),
        elevation: 2,
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 24),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              'Featured Products',
              style: TextStyle(
                fontSize: 24,
                fontWeight: FontWeight.bold,
                color: Color(0xFF1C1B1F),
              ),
            ),
            const SizedBox(height: 6),
            const Text(
              'Hand-picked items just for you',
              style: TextStyle(fontSize: 14, color: Color(0xFF79747E)),
            ),
            const SizedBox(height: 24),
            // ProductCard 1회 사용
            const ProductCard(
              name: 'Wireless Headphones',
              description: 'Premium noise-cancelling audio with 30-hour battery life and foldable design.',
              price: 79.99,
              originalPrice: 129.99,
              imageUrl: 'https://picsum.photos/seed/headphone/400/200',
              badge: 'SALE',
            ),
            // ProductCard 2회 사용
            const ProductCard(
              name: 'Mechanical Keyboard',
              description: 'Tactile switches with RGB backlight, aluminum frame, and USB-C connectivity.',
              price: 149.00,
              imageUrl: 'https://picsum.photos/seed/keyboard/400/200',
              badge: 'NEW',
            ),
            // ProductCard 3회 사용
            const ProductCard(
              name: 'USB-C Hub',
              description: '7-in-1 hub: HDMI 4K, USB 3.0 x3, SD card, PD charging. Plug and play.',
              price: 34.99,
              originalPrice: 49.99,
              imageUrl: 'https://picsum.photos/seed/hub/400/200',
            ),
          ],
        ),
      ),
    );
  }
}

File: fixtures/ios-swiftui-basic/Sources/Screens/DetailScreen.swift
import SwiftUI

/// DetailScreen — 커스텀 컴포넌트 합성 화면
/// ProductCard 커스텀 컴포넌트(내부에 PriceTag 포함 — 2단 깊이)를 3회 사용
struct DetailScreen: View {
    private let products: [ProductItem] = [
        ProductItem(
            title: "Wireless Noise-Cancelling Headphones",
            description: "Premium over-ear headphones with 30-hour battery life and advanced ANC technology.",
            imageName: "product-headphones",
            originalPrice: 299.99,
            discountedPrice: 199.99,
            badge: "SALE"
        ),
        ProductItem(
            title: "Mechanical Keyboard Pro",
            description: "Tactile typing experience with RGB backlight. Compatible with Mac and Windows.",
            imageName: "product-keyboard",
            originalPrice: 149.00,
            discountedPrice: nil,
            badge: "NEW"
        ),
        ProductItem(
            title: "USB-C Hub 7-in-1",
            description: "Expand your laptop ports. Supports 4K HDMI, 100W PD, SD card, and 3x USB-A.",
            imageName: "product-hub",
            originalPrice: 89.99,
            discountedPrice: 69.99,
            badge: nil
        )
    ]

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                Text("Featured Products")
                    .font(.title2)
                    .fontWeight(.bold)
                    .foregroundColor(.primary)
                    .padding(.horizontal, 20)
                    .padding(.top, 16)

                Text("Hand-picked by our team this week")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                    .padding(.horizontal, 20)

                // ProductCard 3회 사용 (커스텀 컴포넌트 합성 검증)
                ForEach(products) { product in
                    ProductCard(
                        title: product.title,
                        description: product.description,
                        imageName: product.imageName,
                        originalPrice: product.originalPrice,
                        discountedPrice: product.discountedPrice,
                        badge: product.badge
                    )
                    .padding(.horizontal, 20)
                }

                Spacer(minLength: 32)
            }
        }
        .navigationTitle("Detail")
        .navigationBarTitleDisplayMode(.large)
        .background(Color(.systemGroupedBackground))
    }
}

// MARK: - 내부 모델

private struct ProductItem: Identifiable {
    let id = UUID()
    let title: String
    let description: String
    let imageName: String
    let originalPrice: Double
    let discountedPrice: Double?
    let badge: String?
}
